Content.Server/Lathe/LatheSystem.cs

```diff
@@ -273,6 +285,16 @@ protected override bool HasRecipe(EntityUid uid, LatheRecipePrototype recipe, La
             return GetAvailableRecipes(uid, component).Contains(recipe.ID);
         }
 
+        private void OnEmagged(EntityUid uid, LatheComponent component, ref GotEmaggedEvent args)
+        {
+            if (!this.IsPowered(uid, EntityManager))
+                return;
+
+            EntityManager.AddComponent<EmaggedComponent>(uid);
+            _popupSystem.PopupEntity(Loc.GetString("lathe-component-got-emmaged"), uid);
+            _audio.PlayPvs(component.EmaggedSound, uid);
```

Comment:
A few things:
- no need to check for being powered. As far as i can tell, most EMAG functions don't require that.
- You don't need to add the EmaggedComponent yourself
- The popup will get played correctly if this event is handled in the shared system